### ISSUE TITLE
Prettier doesn't work if node is not on the PATH

### ISSUE THREAD

**Issue by theodm** [diffplug/spotless#1185]
If you are submitting a **bug**, please include the following:

- [ ] summary of problem
Preface: 
- I want to format my project with prettierjs. 
- NodeJS is not installed in my CI/CD-Environment. 
- I would like to install NodeJS / NPM via node-gradle-plugin. 
- I use the configuration variable .npmExecutable(npmExecutablePath) to select the downloaded npm-executable.

What happens:
The invocation to "npm install" fails with an error code 127. Probably because the npm-executable can't find it's corresponding node-executable. (It's not on the PATH, because it's only downloaded via node-gradle-plugin)

(I manually recreated the environment and the command line result is an error code 127 and the following message: /usr/bin/env: „node“: Datei oder Verzeichnis nicht gefunden. When the node installation directory /bin is on the PATH, it works.)

Probable Workarounds / Fixes:
 - Start npm with the node installation directory as working directory (in https://github.com/diffplug/spotless/blob/3ec902ccd0f7aaf441ca998971c8092d3286c578/lib/src/main/java/com/diffplug/spotless/npm/NpmProcess.java#L57)
 - When starting npm via ProcessBuilder add the node installation directory to the PATH environment variable (in https://github.com/diffplug/spotless/blob/3ec902ccd0f7aaf441ca998971c8092d3286c578/lib/src/main/java/com/diffplug/spotless/npm/NpmProcess.java#L57)
 - Start npm with node-application.
 - Deeper Integration with node-gradle-plugin (https://github.com/diffplug/spotless/issues/728)

- [ ] gradle or maven version
gradle
- [ ] spotless version
6.5.0
- [ ] operating system and version
linux distribution
- [ ] copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
Plugins-Block:
```kotlin
    id("com.diffplug.spotless") version "6.5.0"
    id("com.github.node-gradle.node") version "3.2.1"
```

```kotlin

tasks.npmSetup {
    args.addAll(
        listOf("--registry", "<...>")
    )

    doFirst {
        args.addAll(
            listOf("--registry", "<...>")
        )
    }
}

node {
    workDir.set(rootProject.projectDir.toPath().resolve(".gradle").resolve("nodejs").toFile())
    download.set(true)
    distBaseUrl.set("<...>")

    version.set("16.13.2")
}

tasks.getByName("spotlessInternalRegisterDependencies").dependsOn(":npmSetup")

val npmExecutablePath = VariantComputer()
    .computeNodeDir(node)
    .get()
    .asFile
    .toPath()
    .resolve("bin")
    .resolve("npm")
    .toAbsolutePath()
    .toString()

println(npmExecutablePath)
```

Only a part of the full config:
```kotlin
 java {
        target(
            // Java
            "**/*.java"
        )

        targetExclude(*excludeDirectories.toTypedArray())

        removeUnusedImports()

        prettier(mapOf(prettier, prettierPluginJava))
            .npmExecutable(npmExecutablePath)
            .config(prettierConfig)
    }
```

```kotlin
tasks.getByName("spotlessJava") {
    dependsOn(":npmSetup")
}
```

- [ ] copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`

```
2022-04-25T16:31:44.957+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.957+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
2022-04-25T16:31:44.957+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.957+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':spotlessJava'.
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Running npm command 'npm install --no-audit --no-package-lock' failed with exit code: 127
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Run with --scan to get full insights.
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':spotlessJava'.
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2022-04-25T16:31:44.958+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2022-04-25T16:31:44.959+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: com.diffplug.spotless.npm.NpmProcess$NpmProcessException: Running npm command 'npm install --no-audit --no-package-lock' failed with exit code: 127
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.NpmProcess.npmAwait(NpmProcess.java:50)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.NpmProcess.install(NpmProcess.java:37)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.NpmFormatterStepStateBase.runNpmInstall(NpmFormatterStepStateBase.java:87)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.NpmFormatterStepStateBase.prepareNodeServer(NpmFormatterStepStateBase.java:81)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.NpmFormatterStepStateBase.<init>(NpmFormatterStepStateBase.java:63)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.PrettierFormatterStep$State.<init>(PrettierFormatterStep.java:67)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.npm.PrettierFormatterStep.lambda$create$0(PrettierFormatterStep.java:57)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.FormatterStepImpl.calculateState(FormatterStepImpl.java:56)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.LazyForwardingEquality.state(LazyForwardingEquality.java:56)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.LazyForwardingEquality.toBytes(LazyForwardingEquality.java:85)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at com.diffplug.spotless.LazyForwardingEquality.hashCode(LazyForwardingEquality.java:102)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter$InputCollectingVisitor.visitInputProperty(DefaultInputFingerprinter.java:106)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.TaskExecution.visitRegularInputs(TaskExecution.java:312)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter.fingerprintInputProperties(DefaultInputFingerprinter.java:54)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionStateWithOutputs(CaptureStateBeforeExecutionStep.java:193)
2022-04-25T16:31:44.960+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$captureExecutionState$1(CaptureStateBeforeExecutionStep.java:141)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.BuildOperationStep$1.call(BuildOperationStep.java:37)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.BuildOperationStep.operation(BuildOperationStep.java:34)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionState(CaptureStateBeforeExecutionStep.java:130)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$execute$0(CaptureStateBeforeExecutionStep.java:75)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:75)
2022-04-25T16:31:44.961+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:93)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:93)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.TaskExecution$3.withWorkspace(TaskExecution.java:284)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
2022-04-25T16:31:44.962+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:142)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:131)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2022-04-25T16:31:44.963+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2022-04-25T16:31:44.964+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org/
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] 
2022-04-25T16:31:44.965+0200 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED in 1m 22s
```

If you're just submitting a feature request or question, no need for the above.

**Comment by blutorange** [comment]
Just an idea, perhaps you could also take some inspiration from the [frontend-maven-plugin](https://github.com/eirslett/frontend-maven-plugin)? We're using that to run all our frontend NPM projects from within Maven; and I can say that it "just works" on all machines regardless of what is installed on that machine or what is in the PATH. It uses a local copy of both npm & node.

**Comment by jnels124** [comment]
I opened a [PR](https://github.com/diffplug/spotless/pull/1486) that highlights the issue. If we simply execute npm as a node app node will be known and this error will not occur.

**Comment by simschla** [comment]
This issue should be solved by #1500
